[Do not merge!] Pseudo PR for first release (#31)
nf-core/lsmquant: README.md

## Introduction

**nf-core/lsmquant** is a bioinformatics pipeline that performs preprocessing and analysis of light-sheet microscopy images of tissue cleard samples. The pipeline takes 2D single-channel 16-bit `.tif` images as input. The preprocessing consists of intesity adjustment, channel alignemnt, and tile stitching to reconstruct the 3D image. For mousebrain samples it offers a regsitration to the Allen Mouse Brain Reference Atlas for precise region annotation. Cell nuclei quantification is perfomed on the nuclear channel by a 3D-Unet.

<div style="text-align: center;">
<img src="docs/images/lsmquant-metromap.svg" alt="lasmquant metromap">
</div>

### Basic workflow

**Preprocessing**

1. Intensity Adjustment
2. Channel Alignment
3. Iterative Stitching

**ARA Registration**

4. ARA Registration subworkflow (optional)
5. Cell Nuclei Quantification

**Full**

1. Preprocessing
2. Nuclei quantification

## Pipeline Summary

The pipeline consists of two major workflows `preprocessing` and the `full` workflow. The `ara-regsitration` is an optional subworkflow that works only for whole mouse brain samples.

### Preprocessing

> Preprocessing is performed on raw 2D single-channel 16-bit `.tif` images produced by a light sheet microscope. Three individual steps are perfomed :

Author: ```suggestion
Preprocessing is performed on raw 2D single-channel 16-bit `.tif` images produced by a light sheet microscope. Three individual steps are performed:
```

- **Intensity adjustments** to correct for the Gaussian shape of the lightsheet and intensity differences between adjacent tiles
- **Image channel alignment** using a 2D rigid approach or a nonlinear 3D approach using Elastix.
- **Image tile stitching** via an iterative 2D stitching approach by calculating z displacements and xy translations using phase correlation and SIFT.

### Full

Quantification of cell-nuclei is performed using a 3D-Unet. It is performed on the nuclear channel only, assuming that the corresponding image file names contain the pattern `C1`.

### ARA Registration

> Optionally registration to the Allen Refernce Atlas (ARA) for functional brain region annotation can be perfomed before segmentation.

Author: ```suggestion
Optional registration to the Allen Reference Atlas (ARA) for functional brain region annotation can be perfomed before segmentation.
```
is "registration" here really the right word?

Reviewer: Yes :)

This includes the following two steps:

- Downsampling of the high resolution stitched images
- Registration to the ARA

## Usage

> [!NOTE]
> If you are new to Nextflow and nf-core, please refer to [this page](https://nf-co.re/docs/usage/installation) on how to set-up Nextflow. Make sure to [test your setup](https://nf-co.re/docs/usage/introduction#how-to-run-a-pipeline) with `-profile test` before running the workflow on actual data.

To run the pipeline you need to provide a samplesheet with your data in the following structure:

`samplesheet.csv`

> ```bash

Author: ````suggestion
```csv
````

sample_id,img_directory,parameter_file
TEST1,path/to/image-files,path/to/parameter/file.csv
```

> The parameter csv file includes sample specific parameters that are used for processing the given data.It needs to follow a specific structure.

Author: ```suggestion
The parameter csv file includes sample specific parameters that are used for processing the given data. It needs to follow a specific structure.
```

Author: wait are these parameters for the pipeline processes? I don't think this is the right way to do this

Author: are really all of these different for different samples? If they are I would use yml/json samplesheets instead of csv to make them more readable and keep everything together.

Reviewer: i think many of them can stay the same, but they depend on the imaging procedure, on the different dyes, probably also how the sample preparation was done, the sample it self...  

i can try to change it to yml/json. I think json works better, i think i choose csv because it was less pain within Matlab. 
Just for some Background information: the parameters in the csv file are needed for the Matlab processes. Inside MATLAB i need to create a MATLAB data structure from the parameter csv file, the exact one that is required by the processes otherwise they crash. I thought it was easier to have them in a file instead of all as cli params.

> Please get the basic tempalte file here ( include maybe link to template csv which can be found in the repo ?)

Author: ```suggestion
Please get the basic template file [here](../assets/params_template_lsmquant.csv).
```

[![Get help on Slack](http://img.shields.io/badge/slack-nf--core%20%23lsmquant-4A154B?labelColor=000000&logo=slack)](https://nfcore.slack.com/channels/lsmquant)[![Follow on Bluesky](https://img.shields.io/badge/bluesky-%40nf__core-1185fe?labelColor=000000&logo=bluesky)](https://bsky.app/profile/nf-co.re)[![Follow on Mastodon](https://img.shields.io/badge/mastodon-nf__core-6364ff?labelColor=FFFFFF&logo=mastodon)](https://mstdn.science/@nf_core)[![Watch on YouTube](http://img.shields.io/badge/youtube-nf--core-FF0000?labelColor=000000&logo=youtube)](https://www.youtube.com/c/nf-core)
![HiRSE Code Promo Badge](https://img.shields.io/badge/Promo-8db427?style=plastic&label=HiRSE&labelColor=005aa0&link=https%3A%2F%2Fgo.fzj.de%2FCodePromo)

## Introduction

**nf-core/lsmquant** is a bioinformatics pipeline that performs preprocessing and analysis of light-sheet microscopy images of tissue cleared samples. The pipeline takes raw images from a directory or a zip archive as input. The images need to be in a 2D single-channel 16-bit `tif`format.

<div style="text-align: center;">
<img src="docs/images/lsmquant-metromap.svg" alt="lasmquant metromap">

> </div>

Author: ```suggestion
![lsmquant metromap](docs/images/lsmquant-metromap.svg)
```

Will optimize them automatically on the website. (also good to know for quizes :wink:)

## Pipeline Summary

> The pipeline consists of 3 major components: Preprocessing, Cell-Nuclei quantification, and Allen Reference Atlas registration. A detailed explanation on each method can be found in the [Methods description](./docs/usage.md) section.

Author: ```suggestion
The pipeline consists of 3 major components: Preprocessing, Cell-Nuclei quantification, and Allen Reference Atlas registration. A detailed explanation on each method can be found in the [methods description](./docs/usage#methods-description) section of the usage documentation.
```

**Preprocessing**

This stage reconstructs the 3D image from raw light-sheet data. Here three different workflows can be chosen:

1. `int_align_stitch`: Performs intensity adjustment, channel alignment, and iterative tile stitching

2. `int_stitch`: Performs intensity adjustment and iterative tile stitching.

3. `stitch_only`: Performs only iterative tile stitching

**Allen Brain Atlas Registration (Optional)**

> This workflow registers full brain images to the Allen Brain Reference Atlas. This is an optional workflow and can be chosen by setting the parameter: `ara_registartion`

Author: ```suggestion
This workflow registers full brain images to the Allen Brain Reference Atlas. This is an optional workflow and can be chosen by setting the parameter: [`ara_registration`](./parameters/#ara_registration)
```

> **Cell Nuclei Quantification**

Author: in the initial listing you have this section on second place, ould order them in a similar way here then

> Quantification of cell-nuclei is performed using a 3D-Unet and it is performed on the nuclear channel only. This is an optional workflow and can be chosen by setting the parameter:`nuclei_quantification`

Author: ```suggestion
Quantification of cell-nuclei is performed using a 3D-Unet and it is performed on the nuclear channel only. This is an optional workflow and can be chosen by setting the parameter:[`nuclei_quantification`](./parameters#nuclei_quantification)
```

## Usage

> [!NOTE]
> If you are new to Nextflow and nf-core, please refer to [this page](https://nf-co.re/docs/usage/installation) on how to set-up Nextflow. Make sure to [test your setup](https://nf-co.re/docs/usage/introduction#how-to-run-a-pipeline) with `-profile test` before running the workflow on actual data.

To run the pipeline you need to provide a samplesheet with your data in the following structure:

`samplesheet.csv`

```csv

> sample_id,img_directory,parameter_file

Author: ```suggestion

```csv title="samplesheet.csv"
sample_id,img_directory,parameter_file
```

TEST1,path/to/image-files,path/to/parameter/file.csv
```

The parameter csv file includes sample specific parameters that are used for processing the given data. It needs to follow a specific structure.

Please get the basic template file [here](../assets/params_template_lsmquant.csv).

> `parametersheet.csv`

Author: ```suggestion
```

Now, you can run the pipeline using:

> 

Author: <!--__GRAPHITE_HTML_TAG_START__--><p class='graphite__hidden'><i>[Re: line +74]</i></p><!--__GRAPHITE_HTML_TAG_END__-->
```gt_suggestion
```

<!--__GRAPHITE_HTML_TAG_START__--><p class='graphite__hidden'>See this comment inline on <a href="https://app.graphite.com/github/pr/nf-core/lsmquant/31?utm_source=unchanged-line-comment">Graphite</a>.</p><!--__GRAPHITE_HTML_TAG_END__-->

nf-core/lsmquant was originally written by Carolin Schwitalla.

The pipeline is mainly based on the NuMorph (Nuclear-Based Morphometry) toolbox developed by Krupa et al., 2021.

> **NuMorph: Tools for cortical cellular phenotyping in tissue-cleared whole-brain images**
>
> Krupa O, Fragola G, Hadden-Ford E, Mory JT, Liu T, Humphrey Z, Rees BW, Krishnamurthy A, Snider WD, Zylka MJ, Wu G, Xing L, Stein JL.
>
> Cell Rep. 2021 Oct 12, doi: [10.1016/j.celrep.2021.109802](https://doi.org/10.1016%2Fj.celrep.2021.109802)

We thank the following people for their extensive assistance in the development of this pipeline:

> <!-- TODO nf-core: If applicable, make list of people who have also contributed -->

Author: either add people or remove this line 🙂
